app/controllers/openid_connect/logout_controller.rb

```diff
@@ -25,6 +27,36 @@ def logout
       end
     end
 
+    # rubocop:disable Layout/LineLength
+    #
+    # +POST+ Handle logout request (with confirmation if initiated by relying partner)
+    # @see {OpenID Connect RP-Initiated Logout 1.0 Specification}[https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout]
+    # @note Response status code is <tt>307 Temporary Redirect</tt>
+    #   to preserve the method and body of the request
+    #   and to prevent non-conformant browsers from automatically
+    #   redirecting this route to {OpenidConnectLogoutForm#redirect_uri}
+    #   and skipping over the render of the logout confirmation page
+    #
+    # rubocop:enable Layout/LineLength
+    def create
+      @logout_form = build_logout_form
+      # @type [FormResponse]
+      result = @logout_form.submit
+      redirect_uri = result.extra[:redirect_uri]
+      analytics.oidc_logout_requested(**to_event(result))
+
+      if result.success? && redirect_uri
+        apply_logout_secure_headers_override(redirect_uri, @logout_form.service_provider)
+        if require_logout_confirmation?
+          handle_successful_logout_request(result, redirect_uri, status: :temporary_redirect)
```

Comment:
I see that the `status: :temporary_redirect` is to preserve the method and body of the request, but what makes that necessary?

Reply:
In effect, when testing with `303`, I found Chrome (in particular) would cache the action chosen on the logout confirmation page, so subsequent logout requests would convert the request to a `GET` of the redirect URI, skipping the confirmation page altogether sometimes, which we don't want.

Reply:
The subsequent `GET` request of the redirect URI always happens, but do we always want to skip the confirmation page?

Reply:
Updated this to not use a different status code. The routing behavior I [described](#discussion_r1617622224) was misinformed by a previous implementation that used `match` to send both `POST` and `GET` to the same action.

Reply:
Updated to use a redirect. Removed the functional tests for the `#create` action since it doesn't do anything any more.

Reply:
One thing to keep in mind is that redirecting means we have no analytics around which partners use POST vs GET.

Reply:
We could update our analytics event tracking to log which HTTP verb was used and preserve that info right?

Reply:
@zachmargolis - we would probably need either a new event field (i.e., `original_method`) or a new event type that could be logged in the `#create` method. As far as I can see in my local logs, we don't populate `event_properties.method` any way.

Reply:
New event field makes sense to me!

Reply:
We'd still have the generic web request logs and I don't know if those are bundled into the analytics some other way.

Reply:
those are a separate log file (`production.log`) which doesn't always have SP available, so if we want to analyze HTTP method in `events.log` (by SP it sounds like), then we need to augment analytics events

Reply:
Added the new field. `method` and `original_method` only get populated for the `OIDC Logout Requested` event.

`GET` only:
```jsonc
{
    "name": "OIDC Logout Requested",
    "properties": {
        "event_properties": {
            "success": true,
            "client_id": "urn:gov:gsa:openidconnect:sp:sinatra",
            "client_id_parameter_present": true,
            "id_token_hint_parameter_present": false,
            "errors": {},
            "error_details": null,
            "sp_initiated": true,
            "oidc": true,
            "saml_request_valid": null,
            "method": "GET"
        },
        "new_event": true,
        "path": "/openid_connect/logout",
        "session_duration": 20.555755
        // ...
    },
    // ...
    "log_filename": "events.log"
}
```

`POST` redirected to `GET`:
```jsonc
{
    "name": "OIDC Logout Requested",
    "properties": {
        "event_properties": {
            "success": true,
            "client_id": "urn:gov:gsa:openidconnect:sp:sinatra",
            "client_id_parameter_present": true,
            "id_token_hint_parameter_present": false,
            "errors": {},
            "error_details": null,
            "sp_initiated": true,
            "oidc": true,
            "saml_request_valid": null,
            "method": "GET",
            "original_method": "POST"
        },
        "new_event": false,
        "path": "/openid_connect/logout",
        "session_duration": 781.444997,
        // ...
    },
    // ...
    "log_filename": "events.log"
}
```